GMailSync.txt

```diff
@@ -0,0 +1,34 @@
+Gmail Integration into James
+
+Gmail do that with Gmailify, but it doesn't apply to any IMAP account.
+
+Gmail does synchronization to provide search capabilities, sorting, etc, it's not a backend but rather a bidirectional sync tool.
+
+For James, it makes sense to be able to configure sync processes per user.
+
+It could live outside James process and access James from the outside but it would require talking IMAP or JMAP, non of which is good enough for efficient sync purpose.
+
+Let's consider that the sync process is part of james and uses mailbox-api for now. We could then extract it once mailbox-api would be exposed as REST or RPC.
+
+First, the user will need to define an external account for which it will provide :
+- username
+- OAuth grant rights
+
+Then, the synchronization process should create a new mailbox tree prefixed with external account email address.
```

Comment:
what about using namespaces for that?

Reply:
Well, it would needs right management.

Or we can introduce the fact that ownership is not #private:user:any but follows the pattern any:user:any, which would be of course better. (A bit of work in Cassandra backend...) But I definitly support this!

Reply:
I thought about it but I can't come with a good reason to use namespace. Also, I suspect it would break some IMAP clients.